mobile_verifier/src/reward_shares.rs

```diff
@@ -3,17 +3,132 @@ use crate::{
     speedtests::{Average, SpeedtestAverages},
 };
 use chrono::{DateTime, Duration, TimeZone, Utc};
-use file_store::traits::TimestampEncode;
+use file_store::{mobile_transfer::ValidDataTransferSession, traits::TimestampEncode};
+use futures::{Stream, StreamExt};
 use helium_crypto::PublicKeyBinary;
 use helium_proto::services::{
     follower::{self, follower_gateway_resp_v1::Result as GatewayResult, FollowerGatewayReqV1},
     poc_mobile as proto, Channel,
 };
-use lazy_static::lazy_static;
 use rust_decimal::prelude::*;
+use rust_decimal_macros::dec;
 use std::collections::HashMap;
 use std::ops::Range;
 
+/// 100M genesis rewards per day
+const GENESIS_REWARDS_PER_DAY: Decimal = dec!(100_000_000_000_000);
+
+/// Total emissions pool
+const TOTAL_EMISSIONS_POOL: Decimal = dec!(166_666_666_666_666);
+
+/// Maximum amount of the total emissions pool allocated for data transfer
+/// rewards
+const MAX_DATA_TRANSFER_REWARDS_PERCENT: Decimal = dec!(0.4);
+
+/// The fixed price of a mobile data credit
+const DC_USD_PRICE: Decimal = dec!(0.00000003);
+
+/// Default precision used for rounding
+const DEFAULT_PREC: u32 = 15;
+
+pub struct TransferRewards {
+    scale: Decimal,
+    rewards: HashMap<PublicKeyBinary, Decimal>,
+    pub remaining_poc_rewards: Decimal,
+}
+
+impl TransferRewards {
+    pub fn empty(epoch: &Range<DateTime<Utc>>) -> Self {
+        Self {
+            scale: Decimal::ONE,
+            rewards: HashMap::new(),
+            remaining_poc_rewards: get_scheduled_tokens(epoch.start, epoch.end - epoch.start)
+                .unwrap(),
+        }
+    }
+
+    pub fn scale(&self) -> Decimal {
+        self.scale
+    }
+
+    pub fn reward(&self, hotspot: &PublicKeyBinary) -> Decimal {
+        self.rewards.get(hotspot).copied().unwrap_or(Decimal::ZERO) * self.scale
+    }
+
+    pub async fn from_transfer_sessions(
+        mobile_price: Decimal,
+        transfer_sessions: impl Stream<Item = ValidDataTransferSession>,
+        epoch: &Range<DateTime<Utc>>,
+    ) -> Result<Self, ClockError> {
+        tokio::pin!(transfer_sessions);
+
+        let mut data_transfer_reward_sum = Decimal::ZERO;
+        let rewards = transfer_sessions
+            // Accumulate bytes per hotspot
+            .fold(
+                HashMap::<PublicKeyBinary, Decimal>::new(),
+                |mut entries, session| async move {
+                    *entries.entry(session.pub_key).or_default() +=
+                        Decimal::from(bytes_to_dc(session.download_bytes + session.upload_bytes));
```

Comment:
Shouldn't we be reading the num dcs burned from the valid packet report rather than calculating it again  in the mobile verifier?

Reply:
should we rename `num_dcs` in the packet report then to `num_burnt_dcs`?

Reply:
We confirmed that it is the same calculation so I think `num_dcs` is fine